File: _study/COS_PRO_2/cpp/stage5/2.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

int func_a(vector<int> time_table) {
    int answer = 0;
    for(int i = time_table.size()-1; i >= 0; i--) {
        if(time_table[i] == 1) {
            answer = i;
            break;
        }
    }
    return answer;
}

int func_b(vector<int> time_table, int class1, int class2) {
    int answer = 0;
    for(int i = class1; i < class2; i++)
        if(time_table[i] == 0)
            answer++;
    return answer;
}

int func_c(vector<int> time_table) {
    int answer = 0;
    for(int i = 0; i < time_table.size(); i++) {
        if(time_table[i] == 1) {
            answer = i;
            break;
        }
    }
    return answer;
}

int solution(vector<int> time_table) {
    int answer = 0;
    int first_class = func_c(time_table);
    int last_class = func_a(time_table);
    answer = func_b(time_table, first_class, last_class);
    return answer;
}

// 아래는 테스트케이스 출력을 해보기 위한 main 함수입니다.
int main() {
    vector<int> time_table = {1, 1, 0, 0, 1, 0, 1, 0, 0, 0};
    int ret = solution(time_table);

    // [실행] 버튼을 누르면 출력 값을 볼 수 있습니다.
    cout << "solution 함수의 반환 값은 " << ret << " 입니다." << endl;
}

File: _study/COS_PRO_2/python/stage3/2.py
# -*- coding: utf-8 -*-
# UTF-8 encoding when using korean
def func_a(current_grade, last_grade, rank, max_diff_grade):
    arr_length = len(current_grade)
    count = 0
    for i in range(arr_length):
        if current_grade[i] >= 80 and rank[i] <= arr_length // 10:
            count += 1
        elif current_grade[i] >= 80 and rank[i] == 1:
            count += 1
        elif max_diff_grade > 0 and max_diff_grade == current_grade[i] - last_grade[i]:
            count += 1
    return count

def func_b(current_grade):
    arr_length = len(current_grade)
    rank = [1] * arr_length
    for i in range(arr_length):
        for j in range(arr_length):
            if current_grade[i] < current_grade[j]:
                rank[i] += 1
    return rank

def func_c(current_grade, last_grade):
    max_diff_grade = 1
    for i in range(len(current_grade)):
        max_diff_grade = max(max_diff_grade, current_grade[i] - last_grade[i])
    return max_diff_grade


def solution(current_grade, last_grade):
    rank = func_b(current_grade)
    max_diff_grade = func_c(current_grade, last_grade)
    answer = func_a(current_grade, last_grade, rank, max_diff_grade)
    return answer


current_grade = [70, 100, 70, 80, 50, 95]
last_grade = [35, 65, 80, 50, 20, 60]
ret = solution(current_grade, last_grade)

print("solution 함수의 반환 값은", ret, "입니다.")
